src/sql/src/func.rs

```diff
@@ -3122,12 +3122,22 @@ pub static MZ_CATALOG_BUILTINS: Lazy<BTreeMap<&'static str, Func>> = Lazy::new(|
                     Some(ncols) => ncols,
                 };
                 let ncols = usize::try_from(ncols).expect("known to be greater than zero");
+
+                // Prevent OOMing if the user requests some extremely large number of columns.
+                let mut column_names = Vec::new();
+                if let Err(_) = column_names.try_reserve(ncols) {
```

Comment:
There is a `MAX_STRING_BYTES` const that has a much lower limit for these kinds of things. Recommend we don't allow users to allocate anything allocatable, but instead have a hard coded max.